ospec/ospec.js

```diff
@@ -84,40 +84,56 @@ module.exports = new function init(name) {
 				if (cursor === fns.length) return
 
 				var fn = fns[cursor++]
+				var timeout = 0, delay = 200, s = new Date
+				var isDone = false
```

Comment:
Danger prevents the test suite from running, but I wonder how much time it takes to create these closures unconditionally?

`if (fn.length > 0) {` could become 

`var res; if (fn.length > 0 || (res = fn(), res != null && typeof res.then === 'function;)) {`

And in that if block, assign `done` to a variable and, either call `fn(done)` or `res.then(done)`...

If it doesn't make sense don't worry, I'll have to read this with a fresh mind :-)